packages/moda-sdk/src/profile/getName.ts

```diff
@@ -0,0 +1,38 @@
+import { Address } from "viem";
+import { AddressError, Config, ConfigError } from "../types";
+import { Result } from "../utils/types";
+import { verifyConfig } from "../utils/verifyConfig";
+import { isValidAddress } from "../utils/isValidAddress";
+import { ProfileABI } from "./abi/ProfileABI";
+import { ProfileError } from "./types";
+
+interface GetNameArgs {
+    config: Config;
+    profileAddress: Address;
+}
+
+export async function getName(
+    { config, profileAddress} : GetNameArgs,
+): Promise<Result<string, ConfigError | AddressError | ProfileError>> {
+
+  const verifiedConfig = verifyConfig(config);
+  if (!verifiedConfig.ok) return verifiedConfig;
+
+  const { publicClient } = verifiedConfig.value;
+
+  if (!isValidAddress(profileAddress))
+    return { ok: false, error: AddressError.ProfileAddressInvalid };
+
+  const data = await publicClient.readContract({
+    abi: ProfileABI,
+    address: profileAddress,
+    functionName: 'name',
+  });
+
+ 
+
+  if(data) {
+    return { ok: true, value: data as string };
```

Comment:
Nit: `if (data) return { ok: true, value: data }` on one line is cleaner. 

Also, I'm curious why this can't infer the type as a string via the ABI. we shouldn't have to cast it explicitly. Not the end of the world if we do, but worth seeing why